src/wallet/sqlite.cpp

```diff
@@ -111,6 +111,60 @@ SQLiteDatabase::~SQLiteDatabase()
     }
 }
 
+bool SQLiteDatabase::Verify(bilingual_str& error)
+{
+    if (!PrepareDirectory()) {
+        error = strprintf(_("SQLiteDatabase: Unable to obtain lock on wallet directory \"%s\""), m_dir_path);
+        return false;
+    }
+
+    sqlite3* db = nullptr;
+    int ret = sqlite3_open_v2(m_file_path.c_str(), &db, SQLITE_OPEN_READONLY, nullptr);
```

Comment:
Done.

As a result of this, I've also dropped the usage of `mode` and the database will always be created if it does not exist. This is fine as we enforce existence during `MakeDatabase`, but it would be better to have a `create` flag for this.